Add live indicator to chat tabs

Bauke opened the pull request:
![Screenshot live-indicator](https://user-images.githubusercontent.com/19501722/38950340-71c1a3ac-4345-11e8-8f25-0084b76088f4.png)

Adds a red circle for when the stream is live. 
There's still some issues with it that I'm not sure how to address as I'm fairly new to Angular.

I followed the functionality from [thumbnail.js](https://github.com/mccxiv/tc/blob/master/src/tc-renderer/ng/elements/thumbnail/thumbnail.js) on how to get stream information and passed the streams to `scope.m.streams`. Then used `ng-if="live(channel)"` for the indicator. When Tc first loads this won't work properly so there's a few try/catch blocks to catch the undefined errors. Let me know if there's a better way to do it! (also the circle might be slightly too big, I fine it acceptable but others might not)

Related issue: #370

mccxiv commented:
Great, this is how I would have done it too. Will review over the weekend. 

The only thing I'll change is make the circle smaller and a bit more padding between the channel name. But that's just css

Bauke commented:
Adjusted the size/positioning a bit and looks much better now.

![Live indicator](https://user-images.githubusercontent.com/19501722/39004035-8913210e-43fc-11e8-957e-3e71812a4e5b.png)

Also addressed an issue I found when switching tabs.

mccxiv commented on src/tc-renderer/ng/elements/chat-tabs/chat-tabs.js:
```diff
@@ -120,6 +132,33 @@ angular.module('tc').directive('chatTabs', ($timeout, messages) => {
+    }
   }
 
   return {restrict: 'E', template, link}
```
I've figured out why the values were undefined. It's actually a bug from my part. We need to add `scope: {},` here, so that angularjs creates a new scope for this element instead of using the parent scope

Change to `return {restrict: 'E', scope: {}, template, link}` and then we can be sure that `scope.m.streams` will always exist

mccxiv commented on src/tc-renderer/ng/elements/chat-tabs/chat-tabs.js:
```diff
@@ -120,6 +132,33 @@ angular.module('tc').directive('chatTabs', ($timeout, messages) => {
       arr.splice(toIndex, 0, element)
     }
+
+    async function getStreams() {
```
Once you address the bug below, you can change this to simply

```
async function getStreams () {
      for (const channel of settings.channels) {
        scope.m.streams[channel] = await api.stream(channel)
      }
    }
```

Bauke commented:
Alright everything should be addressed by 2c2e4a49a1d09f3206412d78ec09752af4dbceb2, if there's anything I missed let me know.
